### PR TITLE
8266036: class file for sun.misc.Contended not found

### PR DESCRIPTION
**Pull request by lahodaj**
The ct.sym may contain classfiles referring to annotations that are not present in ct.sym (liek JDK's internal annotation `sun.misc.Contended`). If javac will try to load them (while discovering annotations for the purpose of detecting which annotation processors should be run), an error will be produced (please see the issue). The proposal is to strip annotations that are not present in ct.sym when generating ct.sym.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [ ] Change must be properly reviewed

### Issue
 * [JDK-8266036](https://bugs.openjdk.java.net/browse/JDK-8266036): class file for sun.misc.Contended not found

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/4428/head:pull/4428` \
`$ git checkout pull/4428`

Update a local copy of the PR: \
`$ git checkout pull/4428` \
`$ git pull https://git.openjdk.java.net/jdk pull/4428/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 4428`

View PR using the GUI difftool: \
`$ git pr show -t 4428`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/4428.diff">https://git.openjdk.java.net/jdk/pull/4428.diff</a>

</details>

### CODE REVIEW

**Comment by AlanBateman** [comment]
Seems to be linked to the wrong issue, the issue you want is JDK-8266036.

**Comment by lahodaj** [comment]
> Seems to be linked to the wrong issue, the issue you want is JDK-8266036.

Yes, sorry for the wrong issue. Fixed now.

Thanks for noticing!

File: make/langtools/src/classes/build/tools/symbolgenerator/CreateSymbols.java [hunk 1]
```diff
@@ -301,6 +302,45 @@ public void createSymbols(String ctDescriptionFileExtra, String ctDescriptionFil
         }
     }
 
+    private static final String PREVIEW_FEATURE_ANNOTATION_OLD =
+            "Ljdk/internal/PreviewFeature;";
+    private static final String PREVIEW_FEATURE_ANNOTATION_NEW =
+            "Ljdk/internal/javac/PreviewFeature;";
+    private static final String PREVIEW_FEATURE_ANNOTATION_INTERNAL =
+            "Ljdk/internal/PreviewFeature+Annotation;";
+    public static final Set<String> HARDCODED_ANNOTATIONS = new HashSet<>(
```
Comment (vicente-romero-oracle) [inline make/langtools/src/classes/build/tools/symbolgenerator/CreateSymbols.java:311]: shouldn't PREVIEW_FEATURE_ANNOTATION_INTERNAL be added to HARDCODED_ANNOTATIONS too?

**Comment by jddarcy** [comment]
Does jdk.internal.ValueBased also need special handling? See JDK-8258421.

File: make/langtools/src/classes/build/tools/symbolgenerator/CreateSymbols.java (hunk 1 above)
Comment (lahodaj) [reply make/langtools/src/classes/build/tools/symbolgenerator/CreateSymbols.java:311]: The *_INTERNAL annotation(s) only exists in ct.sym and are injected after the annotations are filtered, so I think it does not need to be in `HARDCODED_ANNOTATION`.

**Comment by lahodaj** [comment]
> Does jdk.internal.ValueBased also need special handling? See JDK-8258421.

@jddarcy, you are right, there needs to be special handling for it.

I have created a new PR for JDK 17 here:
https://github.com/openjdk/jdk17/pull/27

It should also include the special handling for ValueBased and should solve JDK-8258421. If we have more special annotations interpreted directly by javac, we may eventually need a more general way to handle them.

**Pull request closed without merging**